addons/membership/models/partner.py

```diff
@@ -0,0 +1,198 @@
+        string ='Cancel Membership Date', store=True,
+        help="Date on which membership has been cancelled")
+
+    @api.depends('member_lines.account_invoice_line.invoice_id.state',
+                 'member_lines.account_invoice_line.invoice_id.invoice_line_ids',
+                 'member_lines.account_invoice_line.invoice_id.payment_ids',
+                 'free_member',
+                 'membership_state',
+                 'associate_member')
+    def _compute_membership_state(self):
+        values = self._membership_state()
+        for partner in self:
+            partner.membership_state = values[partner.id]
+
+    @api.depends('member_lines.account_invoice_line.invoice_id.state',
+                 'member_lines.account_invoice_line.invoice_id.invoice_line_ids',
+                 'member_lines.account_invoice_line.invoice_id.payment_ids',
+                 'free_member',
+                 'membership_state',
+                 'associate_member')
+    def _compute_membership_start(self):
+        """Return  date of membership"""
+        for partner in self:
+            partner.membership_start = self.env['membership.membership_line'].search([
+                ('partner', '=', partner.associate_member.id or partner.id), ('date_cancel','=',False)
+            ], limit=1, order='date_from').date_from
+
+    @api.depends('member_lines.account_invoice_line.invoice_id.state',
+                 'member_lines.account_invoice_line.invoice_id.invoice_line_ids',
+                 'member_lines.account_invoice_line.invoice_id.payment_ids',
+                 'free_member',
+                 'membership_state',
+                 'associate_member')
+    def _compute_membership_stop(self):
+        MemberLine = self.env['membership.membership_line']
+        for partner in self:
+            partner.membership_stop = self.env['membership.membership_line'].search([
+                ('partner', '=', partner.associate_member.id or partner.id),('date_cancel','=',False)
+            ], limit=1, order='date_to desc').date_to
+
+    @api.depends('member_lines.account_invoice_line.invoice_id.state',
+                 'member_lines.account_invoice_line.invoice_id.invoice_line_ids',
+                 'member_lines.account_invoice_line.invoice_id.payment_ids',
+                 'free_member',
+                 'membership_state',
+                 'associate_member')
+    def _compute_membership_cancel(self):
+        for partner in self:
+            if partner.membership_state == 'canceled':
+                partner.membership_cancel = self.env['membership.membership_line'].search([
+                    ('partner', '=', partner.id)
+                ], limit=1, order='date_cancel').date_cancel
+            else:
+                partner.membership_cancel = False
+
+    def _membership_state(self):
+        """This Function return Membership State For Given Partner. """
+        res = {}
+        today = fields.Date.today()
```

Comment:
A stored function field depending on the current date is probably completely broken